### PR TITLE
Add imputation example

### PR DESCRIPTION
**Pull request by pfistfl**
Had to go over stuff anyway for `robustify`, so I quickly wrote everything up.

### CODE REVIEW

File: content/post/impute-missing-levels.Rmd [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+```
+
+## Prerequisites
+
+This tutorial assumes familiarity with the basics of `r mlr_pkg("mlr3pipelines")`.
+Consult the [mlr3book](https://mlr3book.mlr-org.com/pipelines.html) if some aspects are not  fully understandable.
+It deals with the problem of missing data.
+
+The package `r cran_pkg("ranger")` unfortunately does not support handling of missing features.
+Therefore it is required to manually impute missing features, either in numeric or factor columns.
```
Comment (jakob-r) [inline content/post/impute-missing-levels.Rmd]:
```suggestion
Therefore it is required to impute missing features before passing the data to the learner.
```

File: content/post/impute-missing-levels.Rmd [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+lgr::get_logger("mlr3")$set_threshold("warn")
+```
+
+## Prerequisites
+
+This tutorial assumes familiarity with the basics of `r mlr_pkg("mlr3pipelines")`.
+Consult the [mlr3book](https://mlr3book.mlr-org.com/pipelines.html) if some aspects are not  fully understandable.
+It deals with the problem of missing data.
+
+The package `r cran_pkg("ranger")` unfortunately does not support handling of missing features.
```
Comment (jakob-r) [inline content/post/impute-missing-levels.Rmd]:
```suggestion
The random forest implementation in the package `r cran_pkg("ranger")` unfortunately does not support missing values.
```

File: content/post/impute-missing-levels.Rmd [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+```{r}
+ext_task$missings()
+```
+
+## Putting everything together
+
+Now we have to put all `r ref("PipeOp")`s  together in order to form a graph that handles imputation automatically.
+
+We do this by creating a `r ref("Graph")` that copies the data twice, processes each copy using the respective imputation method and afterwards union the features.
```
Comment (jakob-r) [inline content/post/impute-missing-levels.Rmd]:
```suggestion
We do this by creating a `r ref("Graph")` that copies the data twice, processes each copy using the respective imputation method and afterwards unions the features.
```
Although `union` as verb is rare

File: content/post/impute-missing-levels.Rmd [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+We can now visualize the resulting graph:
+
+```{r}
+graph$plot(html = TRUE)
+```
+
+## Resampling
+
+Correct imputation is especially important when applying imputation to held-out data during the `predict` step.
+If applied incorrectly, imputation could then leak info from the test set, which might invalidate our performance estimates.
```
Comment (jakob-r) [inline content/post/impute-missing-levels.Rmd]:
```suggestion
If applied incorrectly, imputation could leak info from the test set, which potentially skews our performance estimates.
```

File: content/post/impute-missing-levels.Rmd [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+Instead of using `r ref("mlr_pipeops_imputehist", "imputehist")`, we can for example use `r ref("mlr_pipeops_imputenewlvl", "imputenewlvl")`.
+This will simply replace every `NA` in each factor variable with a new value `missing`.
+
+A full graph might the look like this:
+
+```{r}
+imp_fct = po("imputenewlvl")
+graph = po("copy", 2) %>>%
+  gunion(list(imp_missind, imp_num %>>% imp_fct)) %>>%
+  po("featureunion")
```
Comment (jakob-r) [inline content/post/impute-missing-levels.Rmd:197]: Does this really work? Can you apply this graph to some task? It reads as if we can have NAs in factor columns of the test set even without NAs in the training set. But that means that during training ranger never has seen `missing`. How will it handle this new factor for prediction?

**Review by jakob-r — COMMENTED** [review]
Either have a task with a factor column and missing values or leave this topic out. Would be totally ok to just showcase this one aspect and have another example for factor columns.
